servers/src/main/java/tachyon/worker/block/meta/StorageDir.java

```diff
@@ -282,13 +282,14 @@ public void addBlockMeta(BlockMeta blockMeta) throws OutOfSpaceException, Alread
     long blockId = blockMeta.getBlockId();
     long blockSize = blockMeta.getBlockSize();
 
+    if (hasBlockMeta(blockId)) {
+      throw new AlreadyExistsException("Failed to add BlockMeta: blockId " + blockId + " exists");
+    }
```

Comment:
It makes sense to me, but do you want our own exception or simply use the one from preconditions?